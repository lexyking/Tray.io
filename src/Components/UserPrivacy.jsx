import React, { Component } from 'react';


class userPrivacy extends Component {

  //Function continue to move to the next tab
  continue = e => {
    e.preventDefault();
    this.props.nextStep();
  }

  //Function to come back one step
  back = e => {
    e.preventDefault();
    this.props.prevStep();
  }


  render() {
    const label = {
      updateTerm: "Receive updates about Tray.io products by email",
      communication: "Receive Communication by email for other products created by the Tray.io team"
    }
    //Destructuring attributes from the props
    const { handleChange, values } = this.props
    return (
      <section className="mainSection">
        {/* {console.log(values)} */}
        <section className="formSection">
          <h2>Privacy policy</h2>

          <section className="checkBoxes">
            <article>
              <input
                type="checkbox"
                onClick={handleChange}
                name="isUpdate"
                defaultChecked={values.isUpdate}
              />
              <label>{label.updateTerm}</label>
            </article>
            <article>
              <input
                type="checkbox"
                onClick={handleChange}
                name="isCommunication"
                defaultChecked={values.isCommunication}
              />
              <label>{label.communication}</label>
            </article>
          </section>

          <button
            className="btn btn-back"
            onClick={this.back}
          >
            Back
          </button>

          <button
            className="btn"
            onClick={this.continue}
          >
            Continue
          </button>

        </section>
      </section>
    )

  }
}


export default userPrivacy
