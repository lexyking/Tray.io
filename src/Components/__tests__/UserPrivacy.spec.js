import React from 'react';
import { create } from 'react-test-renderer';
import UserPrivacy from '../UserPrivacy';

// Because the header is an invariable component in this applicationCache, we can test the Snapshot
// and compare it to the previous one

describe('UserPrivacy component', () => {
  test('Render Correctly', () => {
    // const values = { isUpdate: false, isCommunication: false }
    const userPrivacy = create(<UserPrivacy values />);
    expect(userPrivacy.toJSON()).toMatchSnapshot();
  })
})