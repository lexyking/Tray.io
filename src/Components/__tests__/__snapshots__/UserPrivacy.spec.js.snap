// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPrivacy component Render Correctly 1`] = `
<section
  className="mainSection"
>
  <section
    className="formSection"
  >
    <h2>
      Privacy policy
    </h2>
    <section
      className="checkBoxes"
    >
      <article>
        <input
          name="isUpdate"
          type="checkbox"
        />
        <label>
          Receive updates about Tray.io products by email
        </label>
      </article>
      <article>
        <input
          name="isCommunication"
          type="checkbox"
        />
        <label>
          Receive Communication by email for other products created by the Tray.io team
        </label>
      </article>
    </section>
    <button
      className="btn btn-back"
      onClick={[Function]}
    >
      Back
    </button>
    <button
      className="btn"
      onClick={[Function]}
    >
      Continue
    </button>
  </section>
</section>
`;
